lib/pedicel/base.rb

```diff
@@ -196,30 +196,21 @@ def self.verify_root_certificate(root:, trusted_root:)
     end
 
     def self.verify_x509_chain(root:, intermediate:, leaf:)
-      store = OpenSSL::X509::Store.new.add_cert(root)
+      # Specifically, ensure that the signature was created using the private
+      # key corresponding to the leaf certificate, that the leaf certificate is
+      # signed by the intermediate CA, and that the intermediate CA is signed by
+      # the Apple Root CA - G3.
```

Comment:
I am sorry in advance for bad questions.

If the `root` parameter is set to something else than `Apple Root CA - G3` (by the user of the method) then we would not literally check that the intermediate CA is signed by `Apple Root CA - G3`, right?

Reply:
That's true, the function `def self.verify_root_certificate(root:, trusted_root:)
` does that.
The comment could be updated :)

Reply:
The comment here is taken from https://developer.apple.com/library/archive/documentation/PassKit/Reference/PaymentTokenJSON/PaymentTokenJSON.html

Reply:
Only "the chain itself" (as "root signed intermediate which in turn signed leaf") is checked. Other stuff is checked elsewhere. The catch here is that we don't want to check that it's the "Apple root CA cert" because we want to be able to use our own CA cert when we test 🙂